ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -738,17 +752,37 @@ shiftUpdate = go
   Extending
 -------------------------------------------------------------------------------}
 
--- | Extend the telescope until the specified slot is within the era at the tip
-extendToSlot :: forall xs. CanHardFork xs
-             => HardForkLedgerConfig xs
-             -> SlotNo
-             -> HardForkState LedgerState xs -> HardForkState LedgerState xs
+-- | Extend the telescope until the specified slot is within the era at the tip.
+--
+-- This function relies on 'singleEraTransition' to find out when a specific era
+-- is ending. If so, we:
+--
+--  * first tick just across the era boundary using the logic of the old era,
+--
+--  * then translate the ledger state into the new era (using
+--    'translateLedgerState').
+--
+-- The final tick to the given slot does /not/ happen here.
```

Comment:
I feel a bit uneasy about the fact that this function has to mention what happens elsewhere. I guess there's no way to avoid it given the current design, but it seems a very delicate piece of logic is scattered around.

Reply:
I could have sworn that I commented in this thread in my Review just now...

My recommendation was ultimately to rename the function `extendToEraOfSlot` or something like that.

Reply:
FTR: You are probably referring to https://github.com/input-output-hk/ouroboros-consensus/pull/340#discussion_r1331794291